Sources/MapboxMapsSwiftUI/MapCoordinator.swift

```diff
@@ -96,3 +106,27 @@ public final class MapCoordinator {
         }.addTo(queriesBag)
     }
 }
+
+private func wrapError(_ body: () throws -> Void) {
+    do {
+        try body()
+    } catch {
+        print("error: \(error)") // TODO: Logger
+    }
+}
+
+private func set<T: Equatable>(_ getter: () -> T, _ setter: (T) throws -> Void, value: T) {
```

Comment:
```suggestion
private func set<T: Equatable>(_ getter: @autoclosure () -> T, _ setter: (T) throws -> Void, value: T) {
```
can this be used with `@autoclosure`? if so, the follow function. `private func set<T: Equatable>(_ oldValue: T, _ setter: (T) throws -> Void, value: T)` can be removed

Reply:
and can we rethrows here and let the callsite decided what do they want to do with errors?

Reply:
Thanks for the idea, we actually don't need a closure for getter at all.

Reply:
Since updateView is called by SwiftUI runtime we don't want any error to be thrown here, so the basic idea is that we catch all those errors and log/display them